eng/tools/TypeSpecValidation/test/tsv-test-host.ts

```diff
@@ -1,57 +1,36 @@
-import { TsvHost } from "../src/tsv-host.js";
+import { IGitOperation, TsvHost } from "../src/tsv-host.js";
 
-export function TsvTestHostGenerator(fnReturnSub: { [key: string]: Object }) {
-  let tsvTestHost = Object.assign({}, DefaultTsvTestHost); // Needs deep copy
-  for (let key in fnReturnSub) {
-    let typedKey = key as keyof TsvHost;
-    if (fnReturnSub[typedKey]) {
-      tsvTestHost = {
-        ...tsvTestHost,
-        [typedKey]: () => {
-          return fnReturnSub[typedKey];
-        },
-      };
-    } else {
-      tsvTestHost = {
-        ...tsvTestHost,
-        [typedKey]: () => {},
-      };
-    }
+export class TsvTestHost implements TsvHost {
```

Comment:
I'd recommend against using class for those use cases. Using pure object makes it much easier to to compose into a new host if needed. 
```ts
{...testHost, runCmd: () => customOverride }
```

but this is also not a major deal, if you prefer it that way. It would matter a lot more I believe if that was consumed by another package.

Reply:
Chatted offline, we agree either approach is OK and I prefer classes.